pkg/metadata/provider_vmware.go

```diff
@@ -0,0 +1,128 @@
+package main
+
+import (
+	"bytes"
+	"compress/gzip"
+	"encoding/base64"
+	"fmt"
+	"io/ioutil"
+	"os/exec"
+	"path"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	guestMetaData = "guestinfo.metadata"
+
+	guestUserData = "guestinfo.userdata"
+)
+
+// ProviderVMware is the type implementing the Provider interface for VMware
+type ProviderVMware struct {
+	cmd string
+}
+
+// NewVMware returns a new ProviderVMware
+func NewVMware() *ProviderVMware {
+	return &ProviderVMware{}
+}
+
+func (p *ProviderVMware) String() string {
+	return "VMWARE"
+}
+
+// Probe checks if we are running on VMware
+func (p *ProviderVMware) Probe() bool {
+	c, err := exec.LookPath("vmware-rpctool")
+	if err != nil {
+		log.Debugf("Look for vmware-rcptool failed %v", err)
```

Comment:
Please don't log in the Probe function. You may get a log messages if not running on VMWare

Reply:
Sorry, it took so long to get back to this.
I fixed the things you suggested.

I changed the probed data as UserData in order to be in more sync with cdrom provider.
Also, I cleaned base64 decoding code a little.

Reply:
@rn I believe the latest patches resolved the issues you pointed.